src/mon/MonClient.cc

```diff
@@ -397,13 +369,13 @@ int MonClient::init()
     return r;
   }
 
-  rotating_secrets = new RotatingKeyRing(cct, cct->get_module_type(), keyring);
+  rotating_secrets.reset(new RotatingKeyRing(
+			   cct, cct->get_module_type(), keyring.get()));
 
   initialized = true;
 
-  timer.init();
   finisher.start();
-  schedule_tick();
+  timer.add_event(tick_time(), &MonClient::timer, this);
```

Comment:
I've been trying to move away from the pattern of having 'init' and 'shutdown' methods separate from constructors and destructors, so if there isn't a compelling reason to do it that way, I'd rather not.

If anything I'd rather throw a patch in that gets rid of MonClient's init and shutdown methods. (But I was being conservative about changes to the interface, apart from removing Context*.)